cpp/serve/request_state.h

```diff
@@ -254,10 +264,15 @@ class RequestStateEntry : public ObjectRef {
 /*! \brief A request's state, which groups all the request state entries. */
 class RequestStateNode : public Object {
  public:
-  /*! \brief the reuqest state entries */
+  /*! \brief the request state entries */
   std::vector<RequestStateEntry> entries;
   /*! \brief tracks the request metrics. */
   RequestMetrics metrics;
+  /*!
+   * \brief The post-process data structures.
+   * We make it a state to avoid repetitive memory allocation/free in the action post process.
+   */
+  RequestActionPostProcStates postproc_states;
```

Comment:
RequestActionPostProcStates => RequestActionPostProcWorkspace

Usually states needs to be persistent across runs, workspace are temp memory that do not need to be persisted